patchwork-extensions-blockentity/src/main/java/net/minecraftforge/common/extensions/IForgeTileEntity.java

```diff
@@ -0,0 +1,197 @@
+	 * sending the packet.
+	 *
+	 * @param net The NetworkManager the packet originated from
+	 * @param pkt The data packet
+	 */
+	default void onDataPacket(ClientConnection net, BlockEntityUpdateS2CPacket pkt) {
+	}
+
+	/**
+	 * Called when the chunk's TE update tag, gotten from {@link #getUpdateTag()}, is received on the client.
+	 *
+	 * <p>Used to handle this tag in a special way. By default this simply calls {@link #readFromNBT(NBTTagCompound)}.
+	 *
+	 * @param tag The {@link NBTTagCompound} sent from {@link #getUpdateTag()}
+	 */
+	default void handleUpdateTag(CompoundTag tag) {
+		getTileEntity().fromTag(tag);
+	}
+
+	/**
+	 * Gets a {@link NBTTagCompound} that can be used to store custom data for this tile entity.
+	 * It will be written, and read from disc, so it persists over world saves.
+	 *
+	 * @return A compound tag for custom data
+	 */
+	CompoundTag getTileData();
+
+	default void onChunkUnloaded() {
+	}
+
+	/**
+	 * Called when this is first added to the world (by {@link World#addTileEntity(TileEntity)}).
+	 * Override instead of adding {@code if (firstTick)} stuff in update.
+	 */
+	default void onLoad() {
+		requestModelDataUpdate();
+	}
+
+	/**
+	 * Return an {@link AxisAlignedBB} that controls the visible scope of a {@link TileEntitySpecialRenderer} associated with this {@link TileEntity}
+	 * Defaults to the collision bounding box {@link Block#getCollisionBoundingBoxFromPool(World, int, int, int)} associated with the block
+	 * at this location.
+	 *
+	 * @return an appropriately size {@link AxisAlignedBB} for the {@link TileEntity}
+	 */
+	// @OnlyIn(Dist.CLIENT)
+	default Box getRenderBoundingBox() {
+		Box bb = INFINITE_EXTENT_AABB;
+		BlockState state = getTileEntity().getCachedState();
+		Block block = state.getBlock();
+		BlockPos pos = getTileEntity().getPos();
+
+		if (block == Blocks.ENCHANTING_TABLE) {
+			bb = new Box(pos, pos.add(1, 1, 1));
+		} else if (block == Blocks.CHEST || block == Blocks.TRAPPED_CHEST) {
+			bb = new Box(pos.add(-1, 0, -1), pos.add(2, 2, 2));
+		} else if (block == Blocks.STRUCTURE_BLOCK) {
+			bb = INFINITE_EXTENT_AABB;
+		} else if (block != null && block != Blocks.BEACON) {
+			Box cbb = null;
+
+			try {
+				cbb = state.getCollisionShape(getTileEntity().getWorld(), pos).getBoundingBox().offset(pos);
+			} catch (Exception e) {
+				// We have to capture any exceptions that may occur here because BUKKIT servers like to send
+				// the tile entity data BEFORE the chunk data, you know, the OPPOSITE of what vanilla does!
+				// So we can not GARENTEE that the world state is the real state for the block...
+				// So, once again in the long line of US having to accommodate BUKKIT breaking things,
+				// here it is, assume that the TE is only 1 cubic block. Problem with this is that it may
+				// cause the TileEntity renderer to error further down the line! But alas, nothing we can do.
+				cbb = new net.minecraft.util.math.Box(pos.add(-1, 0, -1), pos.add(1, 1, 1));
```

Comment:
we're not writing patches

Reply:
=This does not change any vanilla rendering behavior. I would suggest we should keep this. Or simply set cbb to null, so that when an exception is captured, the BE renderer will always be invoked.

Reply:
i meant import instead of `net.minecraft.util.math`

Reply:
fixed